buku.py

```diff
@@ -2565,17 +2565,28 @@ def get_firefox_profile_name(path):
     profile : str
         Firefox profile name.
     """
-
-    try:
-        for name in os.listdir(path):
-            # can be in the format nnnnnn.default or nnnnnn.default-nnnnnn...
-            if '.default' in name and os.path.isdir(os.path.join(path, name)):
-                logdbg(name)
-                return name
-    except FileNotFoundError:
-        pass
-
-    return None
+    from configparser import ConfigParser, NoOptionError
+
+    profile_path = os.path.join(path, 'profiles.ini')
+    if os.path.exists(profile_path):
+        config = ConfigParser()
+        config.read(profile_path)
+        profiles_names = [section for section in config.sections() if section.startswith('Profile')]
+        if not profiles_names:
+            return None
+        for name in profiles_names:
+            try:
+                # If profile is default
+                if config.getboolean(name, 'default'):
+                    profile_path = config.get(name, 'path')
+                    return profile_path
```

Comment:
What if profile_path does not exist?
What if it is not a dir?

This function should check both the cases and return None accordingly.

Reply:
If profile path does not exists -- function returns None
`if os.path.exists(profile_path):`

Reply:
No, you have to add the check:

```
if not os.path.isfile(profile_path):
    return None
```

Reply:
This will take care of both the cases.

Reply:
If `profile_path` points to the folder -- config parser will not return any sections so
`profiles_names = [section for section in config.sections() if section.startswith('Profile')]`  
 will be empty so
```
if not profiles_names:
    return None
```
will finish function

Reply:
OK.